samples/drumthumper/src/main/java/com/google/oboe/sample/drumthumper/DrumPad.kt

```diff
@@ -0,0 +1,212 @@
+//
+//        var numAtts = attrs.attributeCount
+//        var attrName =  ""
+//        var textIndex = -1
+//        for(i in 0..numAtts-1) {
+//            attrName = attrs.getAttributeName(i)
+//            if (attrName == "text") {
+//                textIndex = i
+//                break
+//            }
+//        }
+//
+//        // var styledCound = styleAttrs.indexCount
+//
+//        if (textIndex != -1) {
+//            var chars = attrs.getAttributeValue(3);
+//            mText = chars.toString()
+//        }
+//
+        var xmlns = "http://schemas.android.com/apk/res/android"
+        var textVal = attrs.getAttributeValue(xmlns, "text");
+        if (textVal != null) {
+            mText = textVal;
+        }
+//
+//        var text = styleAttrs.getString(0)
+//        mText = text.toString()
+    }
+
+    //
+    // Layout Routines
+    //
+    private fun calcTextSizeInPixels() : Float {
+
+        val scaledSizeInPixels: Float = TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_SP,
+                mTextSizeSp, resources.displayMetrics)
+
+        return scaledSizeInPixels;
+    }
+    override fun onSizeChanged(w: Int, h: Int, oldw: Int, oldh: Int) {
+        var padLeft = getPaddingLeft()
+        var padRight = getPaddingRight()
+        var padTop = getPaddingTop()
+        var padBottom = getPaddingBottom()
+
+        mDrawRect.set(padLeft.toFloat(),
+                padTop.toFloat(),
+                w - padRight.toFloat(),
+                h - padBottom.toFloat())
+
+        // mTextSize = mDrawRect.bottom / 4.0f
+    }
+
+    override fun onMeasure (widthMeasureSpec: Int, heightMeasureSpec: Int): Unit {
+        var widthMode = MeasureSpec.getMode(widthMeasureSpec);
+        var width = MeasureSpec.getSize(widthMeasureSpec);
+
+//        var padLeft = getPaddingLeft()
+//        var padRight = getPaddingRight()
+        var padTop = getPaddingTop()
+        var padBottom = getPaddingBottom()
+
+
+        var heightMode = MeasureSpec.getMode(heightMeasureSpec);
+        var height = MeasureSpec.getSize(heightMeasureSpec);
+
+        var textSizePixels = calcTextSizeInPixels()
+        when (heightMode) {
+            MeasureSpec.AT_MOST -> run {
+                // mText = "AT_MOST"
+                var newHeight = (textSizePixels.toInt() * 2) + padTop + padBottom
+                height = minOf(height, newHeight) }
+
+            MeasureSpec.EXACTLY -> run {
+                /*mText = "EXACTLY"*/ }
+
+            MeasureSpec.UNSPECIFIED -> run {
+                // mText = "UNSPECIFIED"
+                height = textSizePixels.toInt() }
+        }
+
+        setMeasuredDimension(width, height);
+    }
+
+    //
+    // Drawing Routines
+    //
+    protected override fun onDraw(canvas: Canvas): Unit {
+        // Face
+        if (mIsDown) {
+            mPaint.setColor(mDownColor)
+        } else {
+            mPaint.setColor(mUpColor)
+        }
+        canvas.drawRect(mDrawRect, mPaint)
```

Comment:
Have you experimented with a drawCircle()?
May be quick fun.

Reply:
Looks crappy.